huff_utils/src/sol_interface.rs

```diff
@@ -0,0 +1,98 @@
+///
+/// @param artifacts The vector of artifacts to generate interfaces from.
+/// @return The vector of generated interfaces.
+pub fn gen_sol_interfaces(artifacts: &Vec<Arc<Artifact>>) -> Vec<(PathBuf, &str, String)> {
+    let mut interfaces = Vec::new();
+
+    for artifact in artifacts {
+        if let Some(a) = &artifact.abi {
+            let mut defs = Vec::new();
+            a.events.iter().for_each(|(_, f)| {
+                defs.push(format!(
+                    "{}event {}({});",
+                    "\t",
+                    f.name,
+                    f.inputs
+                        .iter()
+                        .map(|i| {
+                            format!(
+                                "{}{}",
+                                i.kind,
+                                if i.indexed {
+                                    String::from(" indexed")
+                                } else {
+                                    String::default()
+                                }
+                            )
+                        })
+                        .collect::<Vec<_>>()
+                        .join(", "),
+                ));
+            });
+            a.functions.iter().for_each(|(_, f)| {
+                defs.push(format!(
+                    "{}function {}({}) external{}{};",
+                    "\t",
+                    f.name,
+                    f.inputs
+                        .iter()
+                        .map(|i| format!(
+                            "{}{}",
+                            i.kind,
+                            if i.kind.is_memory_type() { " memory" } else { "" }
+                        ))
+                        .collect::<Vec<_>>()
+                        .join(", "),
+                    f.state_mutability.interface_mutability(),
+                    if f.outputs.is_empty() {
+                        String::default()
+                    } else {
+                        format!(
+                            " returns ({})",
+                            f.outputs
+                                .iter()
+                                .map(|o| format!(
+                                    "{}{}",
+                                    o.kind,
+                                    if o.kind.is_memory_type() { " memory" } else { "" }
+                                ))
+                                .collect::<Vec<_>>()
+                                .join(", ")
+                        )
+                    },
+                ));
+            });
+
+            let interface_name =
+                artifact.file.path.split('/').last().unwrap().split('.').next().unwrap();
+            interfaces.push((
+                Path::new(&artifact.file.path).parent().unwrap().to_path_buf(),
+                interface_name,
+                format!("interface I{} {{\n{}\n}}", interface_name, defs.join("\n"),),
+            ));
```

Comment:
Generally don't like lenient unwraps but it really is an error if the source path isn't valid following this pattern

Reply:
Had similar thinking here- in order to reach this code, artifacts must have been successfully generated without error. Doesn't look great, but should be safe since the artifact's file is asserted to exist in `fetch_sources` before the interface gen can run: https://github.com/huff-language/huff-rs/blob/stage/huff_core/src/lib.rs#L109